tss-esapi/src/structures/lists/command_code.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    constants::CommandCode,
+    tss2_esys::{TPM2_MAX_CAP_CC, TPML_CC},
+    Error, Result, WrapperErrorKind,
+};
+use log::error;
+use std::convert::TryFrom;
+
+/// A list of command codes.
+#[derive(Debug, Clone, Default)]
+pub struct CommandCodeList {
+    command_codes: Vec<CommandCode>,
+}
+
+impl CommandCodeList {
+    pub const MAX_SIZE: usize = TPM2_MAX_CAP_CC as usize;
+    /// Creates a new CommandCodeList
+    pub const fn new() -> Self {
+        CommandCodeList {
+            command_codes: Vec::new(),
+        }
+    }
+
+    /// Adds a command code to the command code list.
+    pub fn add(&mut self, command_code: CommandCode) -> Result<()> {
+        if self.command_codes.len() + 1 > CommandCodeList::MAX_SIZE {
+            error!(
+                "Adding command code to list will make the list exceeded its maximum count(> {})",
+                CommandCodeList::MAX_SIZE
+            );
+            return Err(Error::local_error(WrapperErrorKind::WrongParamSize));
+        }
+        self.command_codes.push(command_code);
+        Ok(())
+    }
+
+    /// Returns the number of elements in the list.
+    pub fn len(&self) -> usize {
+        self.command_codes.len()
+    }
+
+    /// Returns true if there are no command codes in the list
+    pub fn is_empty(&self) -> bool {
+        self.command_codes.is_empty()
+    }
```

Comment:
Would this not be covered by the `AsRef` implementation below?

Reply:
How do you mean?

Reply:
Ah, sorry, I confused `AsRef` with `Deref`. That might not actually be a bad idea, actually, or did we agree not to use it for some reason? 🤔

Reply:
I guess Ionut means that is_empty can be executed via AsRef (that returns the inner buffer reference). TBH I don't mind this here, it seems as a useful convenience function 🤷‍♂️

Reply:
It has received Deref now as well.

Reply:
Are the `is_empty` and `len` functions now obsolete since they are Deref'd from the vector?

Reply:
Yes, they are. I removed them.